From 26a14ae3b10d816adf4e4b6d89673514b132cf6b Mon Sep 17 00:00:00 2001
From: croniccorey <cronmod.dev@gmail.com>
Date: Fri, 5 Aug 2016 01:04:37 +0200
Subject: [PATCH 07/13] [aml] AMLUtils: Fix HDMI check

- hpd_state is not always '2' when HDMI is connected but seems
  to always be '0' when HDMI is not connected.
---
 xbmc/utils/AMLUtils.cpp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index 1b54435..6dc74c3 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -277,12 +277,10 @@ bool aml_IsHdmiConnected()
 {
   int hpd_state;
   SysfsUtils::GetInt("/sys/class/amhdmitx/amhdmitx0/hpd_state", hpd_state);
-  if (hpd_state == 2)
-  {
+  if (hpd_state != 0)
     return 1;
-  }
-
-  return 0;
+  else
+    return 0;
 }
 
 bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
-- 
2.7.4

